docs: reorganize root page of Connector Development documentation (#52578)
airbytehq/airbyte: docs/connector-development/README.md

> Welcome to Airbyte's Connector Development guide! This documentation will help you choose the right approach for building your connector and guide you through the development process.

Reviewer: ```suggestion
# Connector Development Guide

Welcome to Airbyte's Connector Development Guide! This documentation will help you choose the right approach for building your connector and guide you through the development process.
```

:::warning Important Note About Java Development

> 🔄 Airbyte is undergoing a major revamp of the Java destinations codebase, with plans to release a new CDK in 2024. Currently:

Reviewer: ```suggestion
🔄 Airbyte is undergoing a major revamp of the Java destinations codebase, with plans to release a new Java/Kotlin CDK in 2025. Currently:
```

# Connector Development Guide

Welcome to Airbyte's Connector Development Guide! This documentation will help you choose the right approach for building your connector and guide you through the development process.

:::warning Important Note About Java Development

> Airbyte is undergoing a major revamp of the Java destinations codebase, with plans to release a new Java/Kotlin CDK in the future. Currently:
> - We are not accepting new Java connector contributions
> - Existing Java connector maintenance continues
> - For new database connectors, please use the Python CDK

Reviewer: ```suggestion
Airbyte is undergoing a major revamp of the Java destinations codebase. Currently:
- We continue to support our certified Java and Kotlin connectors.
- We are not currently accepting community contributions for Java connectors.
```

@ian-at-airbyte - wdyt of this copy rewrite? (Devin: aside)

Reviewer 2: Let's remove one of the two "currently" words. Otherwise I'm pretty happy with this.
